plugins/BEdita/Core/src/Search/Adapter/SimpleAdapter.php

```diff
@@ -0,0 +1,150 @@
+                    $value = (array)$value;
+                    $minLength = Hash::get($config, 'minLength');
+                    $maxWords = Hash::get($config, 'maxWords');
+                    if (count($value) === 0) {
+                        return __d('bedita', 'query strings must be at least {0} characters long', $minLength);
+                    }
+
+                    if ($maxWords > 0 && count($value) > $maxWords) {
+                        return __d('bedita', 'query string too long');
+                    }
+
+                    return true;
+                },
+            ]);
+    }
+
+    /**
+     * Prepare text before search.
+     *
+     * @param string $text The text to search
+     * @param array $options The search options
+     * @param array $config Search configuration
+     * @return array
+     */
+    protected function prepareText(string $text, array $options = [], array $config = []): array
+    {
+        $minLength = Hash::get($config, 'minLength');
+        $words = [$text];
+        if (filter_var(Hash::get($options, 'exact'), FILTER_VALIDATE_BOOLEAN) !== true) {
+            $words = preg_split('/\W+/', $text); // Split words.
+        }
+        $words = array_unique(array_map( // Escape `%` and `\` characters in words.
+            function ($word) {
+                return str_replace(
+                    ['%', '\\'],
+                    ['\\%', '\\\\'],
+                    mb_strtolower($word)
+                );
+            },
+            array_filter( // Filter out words that are too short.
+                $words,
+                function ($word) use ($minLength) {
+                    return mb_strlen($word) >= $minLength;
+                }
+            )
+        ));
+
+        return $words;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function search(Query $query, string $text, array $options = [], array $config = []): Query
+    {
+        $words = $this->prepareText($text, $options, $config);
+        $errors = $this->getValidator($config)->validate(compact('words'));
+        if (!empty($errors)) {
+            throw new BadFilterException([
+                'title' => __d('bedita', 'Invalid data'),
+                'detail' => Hash::get($errors, 'text.0'),
+            ]);
+        }
+
+        $tableFields = (array)Hash::get($config, 'fields');
+
+        // Concat all fields into a single, lower-cased string.
+        $fields = [];
+        /** @var \Cake\ORM\Table $table */
+        $table = $query->getRepository();
+        foreach (array_keys($tableFields) as $field) {
+            $fields[] = $query->func()->coalesce([
+                $table->aliasField($field) => 'identifier',
+                '',
+            ]);
+            $fields[] = ' '; // Add a spacer.
+        }
+        array_pop($fields); // Remove last spacer.
+        $field = new FunctionExpression('LOWER', [$query->func()->concat($fields)]);
```

Comment:
Similarly to my previous consideration, I think wrapping with `LOWER()` is redundant:

```suggestion
        $field = $query->func()->concat($fields);
```